notarization.go

```diff
@@ -48,6 +48,11 @@ func NewNotarization(logger Logger, signatureAggregator SignatureAggregator, vot
 
 	var toBeSignedVote *ToBeSignedVote
 	for _, vote := range votesForCurrentRound {
+		if !bytes.Equal(vote.Vote.Digest[:], blockHeader.Digest[:]) {
```

Comment:
should we keep the same style as your other changes and check the block headers for equality?
```suggestion
		if vote.BlockHeader == blockHeader {
```